### DISCUSSION TITLE
Question regarding Problem 2.2

### DISCUSSION THREAD

**Discussion by into-the-unk** [seominjoon/kaist-ai605#30] [Q&A] [answered]
Hello, professor. 
I have a question regarding Problem 2.2. 
After building an advanced tokenizer that separates non-alphabet characters as independent tokens, I got the same tokenized result as indicated in colab notebook for `hello1 world!!`.
When I check the ratio of exact match, however, I realize that there could be more possible ways to increase this ratio. 
For example, I guess it could be better to have consecutive numbers (for example, `2021`) as a single token instead of separating them as independent tokens. 
I was wondering if it’s okay to add some pre-processing rules to achieve higher exact match ratio?
Thanks in advance :)

**Comment by sunghoonhong** [comment, marked as answer]
I'd like to ask additional question related to this issue.
When I implemented the tokenizer as in notebook, it gave the vocabulary size of about 90k.
I think that too large size of vocabulary can degrade the model performance.
I'm wondering if we should add more preprocessing for reducing the vocabulary size as in Assignment1 (occurence checking).

Bests,

**Reply by seominjoon** [reply]
Yes, absolutely! You can think of it as the minimum requirement, and any additional effort to make it better is welcome.

**Reply by into-the-unk** [reply]
Thank you !
